src/org/zaproxy/zap/extension/ascan/ActiveScanAPI.java

```diff
@@ -532,23 +533,41 @@ private int scanURL(URI url, User user, boolean scanChildren, boolean scanJustIn
 		StructuralNode node;
 		
 		try {
-			node = SessionStructure.find(Model.getSingleton().getSession().getSessionId(), url, method, postData);
+			Session session = Model.getSingleton().getSession();
+			node = SessionStructure.find(session.getSessionId(), url, method, postData);
 			if (node == null) {
 				throw new ApiException(ApiException.Type.URL_NOT_FOUND);
 			}
 			Target target = new Target(node);
 			target.setRecurse(scanChildren);
 			target.setInScopeOnly(scanJustInScope);
+
+			ArrayList<Object> objs = new ArrayList<Object>();
+			if (policy != null) {
+				objs.add(policy);
+			}
+
+			/*
+			 * A side effect of setting the Context based on the specified user
+			 * is the limiting of the scan to technologies associated with the
+			 * Context. When user is not specified, technologies should be
+			 * limited based on Contexts containing the supplied URL.
+			 */
+			TechSet mergedTechSet = new TechSet();
 			if (user != null) {
 				target.setContext(user.getContext());
 			}
-
-			Object [] objs = new Object[]{};
-			if (policy != null) {
-				objs = new Object[]{policy};
+			else {
+				List<Context> contexts = session.getContextsForUrl(url.getURI());
```

Comment:
IIRC in no other place are the context's configurations being merged (it's picked the first context).
This also forces the use of technologies, whether the user wants it or not.